code/modules/antagonists/creep/creep.dm

```diff
@@ -0,0 +1,280 @@
+				spendtime.target = obsessionmind
+				objectives += spendtime
+			if("polaroid")
+				var/datum/objective/polaroid/polaroid = new
+				polaroid.owner = owner
+				polaroid.target = obsessionmind
+				objectives += polaroid
+			if("hug")
+				var/datum/objective/hug/hug = new
+				hug.owner = owner
+				hug.target = obsessionmind
+				objectives += hug
+			if("heirloom")
+				var/datum/objective/steal/heirloom_thief/heirloom_thief = new
+				heirloom_thief.owner = owner
+				heirloom_thief.target = obsessionmind//while you usually wouldn't need this for stealing, we need the name of the obsession
+				heirloom_thief.steal_target = family_heirloom.heirloom
+				objectives += heirloom_thief
+			if("jealous")
+				var/datum/objective/assassinate/jealous/jealous = new
+				jealous.owner = owner
+				jealous.target = obsessionmind//will reroll into a coworker on the objective itself
+				objectives += jealous
+
+	objectives += kill//finally add the assassinate last, because you'd have to complete it last to greentext.
+	for(var/datum/objective/O in objectives)
+		O.update_explanation_text()
+
+/datum/antagonist/creep/roundend_report_header()
+	return 	"<span class='header'>Someone became a creep!</span><br>"
+
+/datum/antagonist/creep/roundend_report()
+	var/list/report = list()
+
+	if(!owner)
+		CRASH("antagonist datum without owner")
+
+	report += "<b>[printplayer(owner)]</b>"
+
+	var/objectives_complete = TRUE
+	if(objectives.len)
+		report += printobjectives(objectives)
+		for(var/datum/objective/objective in objectives)
+			if(!objective.check_completion())
+				objectives_complete = FALSE
+				break
+	if(trauma)
+		if(trauma.total_time_creeping > 0)
+			report += "<span class='greentext'>The [name] spent a total of [DisplayTimeText(trauma.total_time_creeping)] being near [trauma.obsession]!</span>"
+		else
+			report += "<span class='redtext'>The [name] did not go near their obsession the entire round! That's extremely impressive, but you are a shit [name]!</span>"
+	else
+		report += "<span class='redtext'>The [name] had no trauma attached to their antagonist ways! Either it bugged out or an admin incorrectly gave this good samaritan antag and it broke! You might as well show yourself!!</span>"
+
+	if(objectives.len == 0 || objectives_complete)
+		report += "<span class='greentext big'>The [name] was successful!</span>"
+	else
+		report += "<span class='redtext big'>The [name] has failed!</span>"
+
+	return report.Join("<br>")
+
+//////////////////////////////////////////////////
+///CREEPY objectives (few chosen per obsession)///
+//////////////////////////////////////////////////
+
+/datum/objective/assassinate/creep //just a creepy version of assassinate
+
+/datum/objective/assassinate/creep/update_explanation_text()
+	..()
+	if(target && target.current)
+		explanation_text = "Murder [target.name], the [!target_role_type ? target.assigned_role : target.special_role]."
+	else
+		explanation_text = "Free Objective"
```

Comment:
Shouldn't this say something relevant?

Reply:
Honestly it should be telling the admins something, like the trauma not finding an obsession but still getting objectives forged

Reply:
done